cmd/notation/sign.go

```diff
@@ -156,6 +169,59 @@ func prepareSigningContent(ctx context.Context, opts *signOpts, sigRepo notation
 	return signOpts, ref, nil
 }
 
+func signLocal(ctx context.Context, cmdOpts *signOpts, signer notation.Signer, ociImageManifest bool) error {
+	logger := log.GetLogger(ctx)
+
+	mediaType, err := envelope.GetEnvelopeMediaType(cmdOpts.SignerFlagOpts.SignatureFormat)
+	if err != nil {
+		return err
+	}
+	pluginConfig, err := cmd.ParseFlagMap(cmdOpts.pluginConfig, cmd.PflagPluginConfig.Name)
+	if err != nil {
+		return err
+	}
+	userMetadata, err := cmd.ParseFlagMap(cmdOpts.userMetadata, cmd.PflagUserMetadata.Name)
+	if err != nil {
+		return err
+	}
+	var layout ociLayout
+	layout.path, layout.reference, err = parseOCILayoutReference(cmdOpts.reference)
+	if err != nil {
+		return err
+	}
+	sigRepo, err := ociLayoutRepositoryForSign(layout.path, ociImageManifest)
+	if err != nil {
+		return err
+	}
```

Comment:
Looks like this code block is high similar to `prepareRemoteSigningContent`, can we decouple the difference between registry and OCI layout out and use a single prepare function? Most of the difference is on subject reference parsing.